packages/video_player/video_player_avfoundation/ios/Classes/FLTVideoPlayerPlugin.m

```diff
@@ -420,9 +420,15 @@ - (int64_t)duration {
 }
 
 - (void)seekTo:(int)location completionHandler:(void (^)(BOOL))completionHandler {
-  [_player seekToTime:CMTimeMake(location, 1000)
-        toleranceBefore:kCMTimeZero
-         toleranceAfter:kCMTimeZero
+  CMTime locationCMT = CMTimeMake(location, 1000);
+  CMTimeValue duration = _player.currentItem.asset.duration.value;
+  // Without adding tolerance when seeking to duration,
+  // seekToTime will never complete, and this call will hang.
+  // see issue https://github.com/flutter/flutter/issues/124475.
+  CMTime tolerance = location == duration ? CMTimeMake(1, 1000) : kCMTimeZero;
```

Comment:
Can you write a unit test (XCTest) for this behavior?

Reply:
I have no idea what I didn't think of that when @tarrinneal and I were talking about testing for this 🤦🏻 

It looks like we don't currently have a way to mock out the underlying `AVPlayer`, which significantly reduces testability. I would suggest wrapping the current call to `[AVPlayer playerWithPlayerItem:]` in a factory object, and DI of that factory. So you'd make a test-only header that declared:
- A protocol for the `AVPlayer` instance factory (which it looks like would just be a single method)
- A new init that takes an instance of the factory protocol.

The, similar to the other DI you did recently, you'd have a private implementation of that protocol that's the default version, just wrapping `[AVPlayer playerWithPlayerItem:]`, but tests could replace it with something vending mock AVPlayer instances where you could validate the tolerance parameter.

Reply:
Can also try stubbing the constructor directly if it's easier: 

```
id mockPlayer = OCMClassMock([AVPlayer class]);
OCMStub([AVPlayer playerWithPlayerItem:OCMOCM_ANY]).andReturn(mockPlayer);
```

Reply:
Please don't. OCMock class stubbing is giant foot-gun; it's incredibly easy to accidentally leak mocking across tests.

Reply:
gotcha. i did that a few times for camera plugin iirc. i should clean that up sometime.

Reply:
Does this implementation look correct? No tests yet.

Reply:
Added the test, but I'm not sure how to access the tolerance within the completion handler, since I can't change the parameters